nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/JCAMPData.java

```diff
@@ -1056,58 +678,79 @@ public void setSampleSchedule(SampleSchedule sampleSchedule) {
         this.sampleSchedule = sampleSchedule;
     }
 
-    class BrukerSinebellWt extends SinebellWt {
+    @Override
+    public DatasetType getPreferredDatasetType() {
+        return preferredDatasetType;
+    }
 
-        BrukerSinebellWt(int iDim) {
-            Integer wdw = getParInt("WDW," + (iDim + 1));
-            String spar;
-            if (wdw != null && (wdw == 3 || wdw == 4)) {
-                if ((spar = getPar("SSB," + (iDim + 1))) != null) {
-                    if (!spar.equals("n")) {
-                        if (wdw == 4) {
-                            power = 2;
-                        } else {
-                            power = 1;
-                        }
-                        sb = 1.0;
-                        sbs = Double.parseDouble(spar);
-//                      size = -1; // size = "";
-                        if (sbs < 2.0) {
-                            offset = 0.0;
-                        } else {
-                            offset = 1.0 / sbs;
-                        }
-                        end = 1.0;
-                    }
-                }
-            }
-        }
+    @Override
+    public void setPreferredDatasetType(DatasetType datasetType) {
+        this.preferredDatasetType = datasetType;
+    }
 
+    @Override
+    public boolean getNegatePairs(int dim) {
+        //XXX doesn't seem to be called. Is this useful?
+        return "negate".equals(getFTType(dim));
     }
 
-    class BrukerGaussianWt extends GaussianWt {
+    @Override
+    public boolean getNegateImag(int dim) {
+        //XXX doesn't seem to be called. Is this useful?
+        return dim > 0;
+    }
 
-        BrukerGaussianWt(int iDim) {
-            Integer wdw = getParInt("WDW," + (iDim + 1));
-            String spar;
-            if (wdw != null && wdw == 2) {
-                if ((spar = getPar("GB," + (iDim + 1))) != null) {
-                    if (!spar.equals("n")) {
-                        gf = Double.parseDouble(spar);
-                        if ((spar = getPar("LB," + (iDim + 1))) != null) {
-                            if (!spar.equals("n")) {
-                                lb = Double.parseDouble(spar);
-                            }
-                        }
-                    }
-                }
-            }
-        }
+    @Override
+    public long getDate() {
+        return block.getDate().getTime() / 1000;
+    }
 
+    @Override
+    public boolean isFID() {
+        return block.getDataType().isFID();
     }
 
-    public static void main(String[] args) {
+    @Override
+    public String toString() {
+        return getFilePath();
+    }
 
+    private FnMode getFnMode(int dim) {
+        int value = block.optional($FN_MODE, dim).map(JCampRecord::getInt).orElse(-1);
+        if(value < 0 || value >= FnMode.values().length)
+            return null; // XXX should really be "no FnMode defined" be different from FnMode == "0/undefined"?
```

Comment:
The original and bruker parsers have different behavior for FnMODE not defined and FnMODE = 0 (undefined).